api/v1alpha3/docluster_types.go

```diff
@@ -50,6 +57,14 @@ type DOClusterStatus struct {
 	Network DONetworkResource `json:"network,omitempty"`
 }
 
+type DOControlPlaneDNSRecord struct {
+	// Domain is the DO domain that this record should live in.
+	// It must be pre-existing in your DO account.
+	Domain string `json:"domain"`
+	// Name is the DNS short name of the record (non-FQDN)
+	Name string `json:"name"`
```

Comment:
What happened when we fill this `name` with _@_ sign? will it use the root domain?

Reply:
This half-works and falls apart later. Do we really want this to work?
Should I explicitly prohibit it?

Reply:
Added a check to disallow `*` and `@`. We can always allow them later should someone have a proper use-case for them.

Reply:
I did not try to fill this with `@` sign before. And yes, let's disallow it for now.